lib/workspace.js

```diff
@@ -245,16 +245,28 @@ module.exports = {
             }
         } else if (workspace.homedir_location == 'FileSystem') {
             const root = config.workspaceHomeDirRoot;
+            const fullRemotePath = path.join(root, remotePath);
+            const timestampSuffix = (new Date()).toISOString().substring(0,19).replace(/[^a-zA-Z0-9]/g, '-');
+            try {
+                // Move any existing directory out of the way to get a clean start.
```

Comment:
Possibly overkill, but IMO detailed comments around tricky issues like this are always helpful.

```suggestion
                // Move any existing directory out of the way to get a clean start.
                // This should never happen in production environments, but when
                // running workspaces locally in development, we may end up trying
                // to reuse the same workspace ID and thus directory, for intstance
                // if the database is reset in the middle of testing. In that case,
                // we want to ensure that we don't try to write on top of an existing
                // directory, as this could have unexpected outcomes.
```

Reply:
Updated this comment in 62fd331.